document-store/src/main/java/org/hypertrace/core/documentstore/expression/impl/AliasedIdentifierExpression.java

```diff
@@ -0,0 +1,30 @@
+package org.hypertrace.core.documentstore.expression.impl;
+
+import com.google.common.base.Preconditions;
+import lombok.Value;
+
+/**
+ * Expression representing an identifier/column name with an alias
+ *
+ * <p>Example: AliasedIdentifierExpression.of("col1", "col1_alias");
+ */
+@Value
+public class AliasedIdentifierExpression extends IdentifierExpression {
+  String alias;
+
+  private AliasedIdentifierExpression(final String name, final String alias) {
+    super(name);
+    this.alias = alias;
+  }
+
+  public static AliasedIdentifierExpression of(final String name, final String alias) {
```

Comment:
Direct/static constructors like this with multiple arguments of the same datatype are error-prone since the callers can accidentally pass them in incorrect order.

Please prefer a builder instead.